### PR TITLE
Benchmarks: Refactor benchmarks to be provider agnostic

### LINKED ISSUE #9681: Benchmarks: Run with SQLite provider
This will help us detect any regressions in both Microsoft.EntityFrameworkCore.Sqlite and Microsoft.Data.Sqlite.

(For both 2.0.0 and dev)

### PR DESCRIPTION
**Pull request by smitpatel**
Run benchmarks for Sqlite provider

Resolves #9681

### CODE REVIEW

**Comment by smitpatel** [comment]
cc: @roji

File: benchmarks/test/EFCore.Benchmarks.Sqlite.Dev/Models/AdventureWorks/AdventureWorksFixture.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.EntityFrameworkCore.Benchmarks.Models.AdventureWorks
+{
+    public static class AdventureWorksFixture
+    {
+        // TODO: No database for Sqlite
+        private static string _connectionString = $"DataSource=AdventureWorks2014;";
```
Comment (smitpatel) [inline benchmarks/test/EFCore.Benchmarks.Sqlite.Dev/Models/AdventureWorks/AdventureWorksFixture.cs]: @bricelam some of our tests uses AdventureWorks database. To run those tests, we would need copy of the database for Sqlite.

File: benchmarks/test/EFCore.Benchmarks.Sqlite.Dev/Query/RawSqlQuerySqliteTests.cs [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+using Microsoft.EntityFrameworkCore.Benchmarks.Models.Orders;
+
+namespace Microsoft.EntityFrameworkCore.Benchmarks.Query
+{
+    public class RawSqlQuerySqliteTests : RawSqlQueryTests
+    {
+        protected override string StoredProcedureCreationScript
+            => @"";
+
+        // TODO: Define stored procedure creation script
```
Comment (smitpatel) [inline benchmarks/test/EFCore.Benchmarks.Sqlite.Dev/Query/RawSqlQuerySqliteTests.cs]: @bricelam how to define a stored procedure in sqlite? We have definition for SqlServer right now. It is used in a test where FromSql is used with sproc

File: benchmarks/test/EFCore.Benchmarks.Sqlite.Dev/Models/AdventureWorks/AdventureWorksFixture.cs (hunk 1 above)
Comment (bricelam) [reply benchmarks/test/EFCore.Benchmarks.Sqlite.Dev/Models/AdventureWorks/AdventureWorksFixture.cs]: As discussed offline, use `.EnsureCreated()` to create the schema, select the data from SQL Server and insert into SQLite.

File: benchmarks/test/EFCore.Benchmarks.Sqlite.Dev/Query/RawSqlQuerySqliteTests.cs (hunk 2 above)
Comment (bricelam) [reply benchmarks/test/EFCore.Benchmarks.Sqlite.Dev/Query/RawSqlQuerySqliteTests.cs]: No procedures in SQLite--just TVFs.

File: benchmarks/test/EFCore.Benchmarks.Sqlite.Dev/Models/AdventureWorks/AdventureWorksFixture.cs (hunk 1 above)
Comment (smitpatel) [reply benchmarks/test/EFCore.Benchmarks.Sqlite.Dev/Models/AdventureWorks/AdventureWorksFixture.cs]: Can you file an issue for that?

Comment (bricelam) [reply benchmarks/test/EFCore.Benchmarks.Sqlite.Dev/Models/AdventureWorks/AdventureWorksFixture.cs]: #11120

**Pull request merged**